### PR TITLE
service/apt: Add and implement more service commands.

### PR DESCRIPTION
**Pull request by Steveice10**
* Defines and wires up the full New 3DS hardware capabilities from the ARM11 Local Capabilities in the exheader to the kernel, so the parameters can be utilized.
* Fixes up some newer APT command names based on findings from Fire Emblem Fates.
* Implements a bunch of relatively low hanging fruit in the APT module, including: Finalize, CountRegisteredApplet, GetAttribute, CancelApplication, SendDeliverArg, SendDspSleep, SendDspWakeUp, GetProgramInfo, Reboot, HardwareResetAsync, WakeupApplication2, GetProgramId, GetApplicationRunningMode, IsStandardMemoryLayout
* Implements a power button, which allows you to shut down within the home menu. It is disabled unless the home menu was booted by the user.
* Makes better use of utility functions in APT, such as those provided by AM for titles.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/6721)

### CODE REVIEW

**Comment by Steveice10** [comment]
This should be ready now.

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -1154,37 +1331,93 @@ void Module::APTInterface::Unwrap(Kernel::HLERequestContext& ctx) {
...
+        rb.PushEnum(info.Unwrap());
+    }
+}
+
+void Module::APTInterface::IsStandardMemoryLayout(Kernel::HLERequestContext& ctx) {
+    IPC::RequestParser rp(ctx);
+
+    LOG_DEBUG(Service_APT, "called");
+
+    bool standard;
```
Comment (FearlessTobi) [inline src/core/hle/service/apt/apt.cpp:1408]:
```suggestion
    bool is_standard;
```

File: src/core/hle/service/apt/apt.h [hunk 2]
```diff
@@ -907,7 +1053,7 @@ class Module final {
     u32 cpu_percent = 0; ///< CPU time available to the running application
 
     // APT::CheckNew3DSApp will check this unknown_ns_state_field to determine processing mode
```
Comment (FearlessTobi) [inline src/core/hle/service/apt/apt.h:1055]: comment needs to be updated

**Review by FearlessTobi — APPROVED** [review]

File: src/core/hle/service/apt/applet_manager.cpp [hunk 3]
```diff
@@ -1286,12 +1457,12 @@ AppletManager::AppletManager(Core::System& system) : system(system) {
...
-    home_button_update_event = Core::System::GetInstance().CoreTiming().RegisterEvent(
-        "Home Button Update Event", [this](std::uintptr_t user_data, s64 cycles_late) {
-            HomeButtonUpdateEvent(user_data, cycles_late);
+    button_update_event = Core::System::GetInstance().CoreTiming().RegisterEvent(
+        "APT Button Update Event", [this](std::uintptr_t user_data, s64 cycles_late) {
+            ButtonUpdateEvent(user_data, cycles_late);
         });
-    Core::System::GetInstance().CoreTiming().ScheduleEvent(
-        usToCycles(home_button_update_interval_us), home_button_update_event);
+    Core::System::GetInstance().CoreTiming().ScheduleEvent(usToCycles(button_update_interval_us),
```
Comment (GPUCode) [inline src/core/hle/service/apt/applet_manager.cpp:1464]: Unrelated to the PR specifically but the class has system as a member already, so no need to rely on the global instance here. There are a few other cases of this

**Review by GPUCode — APPROVED** [review]

File: src/core/hle/service/apt/applet_manager.cpp (hunk 3 above)
Comment (Steveice10) [reply src/core/hle/service/apt/applet_manager.cpp]: Removed all but one, which is removed entirely by my other PR related to framebuffer capture.

**Pull request merged**